src/Components/Tasks/Tasks.tsx

```diff
@@ -1,129 +1,216 @@
+      const doneTask=response?.data?.data.filter((task:any)=>task.status=='Done')
+      setTodos(todosTask)
+      setInProgress(inProgressTask)
+      setDone(doneTask)
+  
+      setTasks(response.data.data)
+   
+    })
+   }
+  
 
   useEffect(() => {
+    if(role==='Manager')
     getTasksList();
+  else
+  getEmployeeTasks()
   }, []);
 
+
+  const statuses =['ToDo','InProgress','Done']
   return (
     <>
-         <Modal show={modalState==='delete-modal'&& "update-modal"} onHide={handleClose}>
+  
+      <Modal show={modalState === 'delete-modal'} onHide={handleClose}>
 
-         <Modal.Body>
-       <div className="delete-container">
-          <div className="icons text-end">
-        <i onClick={handleClose} className="fa-regular fa-circle-xmark text-danger "></i>
-          </div>
-          <div className="text-center">
+        <Modal.Body>
+          <div className="delete-container">
+            <div className="icons text-end">
+              <i onClick={handleClose} className="fa-regular fa-circle-xmark text-danger "></i>
+            </div>
             <div className="text-center">
-            <img className=' '  src={noData} alt="msg-NoData" />
+              <div className="text-center">
+                <img  src={Datano} alt="msg-NoData" />
+              </div>
+              <h5 className='py-3'> Are you sure to Delete this item ? </h5>
             </div>
-            <h5 className='py-3'> Are you sure to Delete this item ? </h5>
-          </div>
 
-          <div className="delete-btn text-end">
-            <button  className='text-white bg-danger btn btn-outline-danger   border-danger p-3  '>Delete This Item </button>
+            <div className="delete-btn text-end">
+              <button onClick={deleteTask} className='text-white bg-danger btn btn-outline-danger   border-danger p-3  '>Delete This Item </button>
+            </div>
           </div>
+        </Modal.Body>
+      </Modal>
+      <div className="container bg-white rounded-3 px-3">
+        <div className="col-md-12 trans-head">
+          <div className="top d-flex justify-content-between p-3">
+            <h3>Tasks</h3>
+            {role == "Manager" ? <button onClick={navToAddTask} className='btn bg-warning text-white'> <i className='fa-plus'></i> Add New Task </button>
+              : ""}
+            
+          </div>
+
         </div>
-       </Modal.Body>
-       </Modal>
-       <div className="container bg-white rounded-3 px-3">
-  <div className="col-md-12 trans-head">
-    <div className="top d-flex justify-content-between p-3">
-    <h3>Tasks</h3>
-        {role=="Manager"?<button onClick={addToTask} className='btn bg-warning text-white'> <i className='fa-plus'></i> Add New Task </button>
-:""}
-    </div>
+        <div className="col-md-12 ">
+          <input type="text" placeholder='SearchFleets ' className='rounded-4 border-1 mb-4 pro' />
```

Comment:
Efficiently managing user input, especially in search functionality, is pivotal for a seamless user experience. 🚀

In your code, `debouncing` is a smart addition. `Debouncing` prevents excessive API calls by introducing a slight delay before executing the actual search function. This is beneficial in scenarios where users may rapidly type or change their search query. 🕒

Here's a concise guide with a friendly nudge to explore and implement your own debouncing approach:

1. Understanding Debouncing: 🧠
> Debouncing involves introducing a delay before executing a function after a user action, like typing. This delay ensures that the function is only invoked when there's a brief pause in user activity.

2. Debouncing Function (Custom Implementation): 🛠️
> While `lodash` provides a `debounce` function, consider crafting your own `debounce` function for a deeper understanding. A simple `debounce` function looks like this:
```jsx
const debounce = (callback, delay) => {
  let timeoutId;
  return (...args) => {
    clearTimeout(timeoutId);
    timeoutId = setTimeout(() => {
      callback(...args);
    }, delay);
  };
};
```
3. Debouncing Search Input in Component: 🎛️
> Apply debouncing to your search input. In your case, the snippet can look like this:
```jsx
const [debouncedSearchInput, setDebouncedSearchInput] = useState('');
const debounceDelay = 500; // Adjust the delay as needed

const debouncedSetSearchInput = debounce((value) => {
  setDebouncedSearchInput(value);
  // Optionally, trigger your search function here
}, debounceDelay);
```
4. Integration with `useEffect`: 🔄
> Leverage the `debouncedSetSearchInput` within a `useEffect` that watches changes in the original search input:
```jsx
useEffect(() => {
  debouncedSetSearchInput(searchInput);
  // Optionally, perform other actions or trigger API calls
}, [searchInput]);
```
Remember, the joy of coding lies in exploration and understanding. Feel free to customize and experiment with `debounce` logic to suit your application's unique requirements! 🚀👩‍💻